### PR TITLE
chore: set up pre-commit hooks (DEV-2393)

### PR DESCRIPTION
**Pull request by jnussbaum**
resolves DEV-2393

### CODE REVIEW

File: testdata/bitstreams/test.csv [hunk 1]
```diff
@@ -1,4 +1,4 @@
-﻿artwork;;
+artwork;;
```
Comment (jnussbaum) [inline testdata/bitstreams/test.csv:1]: remove UTF-8 Byte Order Mark (made by one of the pre-commit hooks)

File: testdata/excel2xml/excel2xml-testdata.csv [hunk 2]
```diff
@@ -1,4 +1,4 @@
...
+id,restype,label,ark,iri,created,permissions,file,file permissions,prop name,prop type,prop list,1_value,1_encoding,1_permissions,1_comment,2_value,2_encoding,2_permissions,2_comment,3_value,3_encoding,3_permissions,3_comment,4_value,4_encoding,4_permissions,4_comment,           ,,,,,,          ,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,,
```
Comment (jnussbaum) [inline testdata/excel2xml/excel2xml-testdata.csv:1]: remove UTF-8 Byte Order Mark (made by one of the pre-commit hooks)

File: testdata/excel2xml/excel2xml-testdata.csv [hunk 3]
```diff
@@ -24,18 +24,18 @@ region_0,Region,Region of img_obj_6,,http://rdfh.ch/4123/54SYvWF0QUW6a,,res-rest
...
+                <a class=""salsah-link"" href=""IRI:test_thing_1:IRI"">test_thing_1</a>
+                <a class=""salsah-link"" href=""IRI:image_thing_0:IRI"">image_thing_0</a>
+                <a class=""salsah-link"" href=""IRI:compound_thing_0:IRI"">compound_thing_0</a>
+                <a class=""salsah-link"" href=""IRI:partof_thing_1:IRI"">partof_thing_1</a>
+                <a class=""salsah-link"" href=""IRI:partof_thing_2:IRI"">partof_thing_2</a>
+                <a class=""salsah-link"" href=""IRI:partof_thing_3:IRI"">partof_thing_3</a>
+                <a class=""salsah-link"" href=""IRI:document_thing_1:IRI"">document_thing_1</a>
+                <a class=""salsah-link"" href=""IRI:text_thing_1:IRI"">text_thing_1</a>
+                <a class=""salsah-link"" href=""IRI:zip_thing_1:IRI"">zip_thing_1</a>
+                <a class=""salsah-link"" href=""IRI:audio_thing_1:IRI"">audio_thing_1</a>
```
Comment (jnussbaum) [inline testdata/excel2xml/excel2xml-testdata.csv:38]: Don't know what the tool made here, but I compared the diff carefully, and there isn't a semantic change.

**Review by BalduinLandolt — APPROVED** [review]
How long does this hook run? Personally I would be very much opposed to a workflow that makes committing take more than 5-10 seconds. But of course it's up to you in the end.

**Comment by jnussbaum** [comment]
It only takes ca. 3-5 seconds. I think I will collect some experience, and see if I'm going to like it or not.

**Pull request merged**